pkg/scanners/scannerv4/nodescan_convert.go

```diff
@@ -9,144 +9,79 @@ import (
-		OperatingSystem: convertedOS,
+		ScanTime:        protocompat.TimestampNow(),
+		OperatingSystem: nodeOS(osImage),
+		Components:      nodeComponents(vr),
+		Notes:           nodeNotes(vr, osImage),
 	}
 }
 
-func toOperatingSystem(ref string) string {
-	r := rhcosOSImageRegexp.FindStringSubmatch(ref)
+func nodeOS(osImage string) string {
+	r := rhcosOSImagePattern.FindStringSubmatch(osImage)
 	if len(r) != 5 {
-		return ""
+		return "unknown"
 	}
 	return fmt.Sprintf("rhcos:%s.%s", r[3], r[4])
 }
 
-func toStorageComponents(r *v4.VulnerabilityReport) []*storage.EmbeddedNodeScanComponent {
-	packages := r.GetContents().GetPackages()
-	result := make([]*storage.EmbeddedNodeScanComponent, 0, len(packages))
-
-	for _, pkg := range packages {
-		vulns := getPackageVulns(pkg.GetId(), r)
-		result = append(result, createEmbeddedComponent(pkg, vulns))
-	}
-	return result
-}
-
-func getPackageVulns(packageID string, r *v4.VulnerabilityReport) []*storage.EmbeddedVulnerability {
-	vulns := make([]*storage.EmbeddedVulnerability, 0)
-	mapping, ok := r.GetPackageVulnerabilities()[packageID]
-	if !ok {
-		// No vulnerabilities for this package, skip
-		return vulns
-	}
-	processedVulns := set.NewStringSet()
-	for _, vulnID := range mapping.GetValues() {
-		if !processedVulns.Add(vulnID) {
-			// Already processed this vulnerability, skip it
-			continue
-		}
-		vulnerability, ok := r.Vulnerabilities[vulnID]
-		if !ok {
-			log.Debugf("Mapping for package %s contains a vulnerability with unknown ID %s. This vulnerability won't be stored", packageID, vulnID)
-			continue
-		}
-		vulns = append(vulns, convertVulnerability(vulnerability))
-	}
-	return vulns
-}
-
-func convertVulnerability(v *v4.VulnerabilityReport_Vulnerability) *storage.EmbeddedVulnerability {
-	converted := &storage.EmbeddedVulnerability{
-		Cve:               v.GetName(),
-		Summary:           v.GetDescription(),
-		VulnerabilityType: storage.EmbeddedVulnerability_NODE_VULNERABILITY,
-		Severity:          normalizedSeverity(v.GetNormalizedSeverity()),
-		Link:              link(v.GetLink()),
-		PublishedOn:       v.GetIssued(),
-	}
-
-	if err := setScoresAndScoreVersions(converted, v.GetCvssMetrics()); err != nil {
-		utils.Should(err)
-	}
-	maybeOverwriteSeverity(converted)
-	if v.GetFixedInVersion() != "" {
-		converted.SetFixedBy = &storage.EmbeddedVulnerability_FixedBy{
-			FixedBy: v.GetFixedInVersion(),
+func nodeComponents(report *v4.VulnerabilityReport) []*storage.EmbeddedNodeScanComponent {
+	pkgs := report.GetContents().GetPackages()
+	components := make([]*storage.EmbeddedNodeScanComponent, 0, len(pkgs))
+	for _, pkg := range pkgs {
+		id := pkg.GetId()
+		vulnIDs := report.GetPackageVulnerabilities()[id].GetValues()
+
+		component := &storage.EmbeddedNodeScanComponent{
+			Name:    pkg.GetName(),
+			Version: pkg.GetVersion(),
+			Vulns:   vulnerabilities(vulnIDs, report.GetVulnerabilities(), storage.EmbeddedVulnerability_NODE_VULNERABILITY),
```

Comment:
In the past this would have been set to `[]*storage.EmbeddedVulnerability` if were no vulns found, now it will be set to `nil` - will that break anything in the UI / other workflows?